src/orange/BoundingBox.hh

```diff
@@ -7,65 +7,81 @@
 //---------------------------------------------------------------------------//
 #pragma once
 
-#include <cmath>
+#include <type_traits>
 
 #include "corecel/Assert.hh"
 #include "corecel/Macros.hh"
 #include "corecel/cont/Array.hh"
 #include "corecel/math/NumericLimits.hh"
 
-#include "Types.hh"
+#include "OrangeTypes.hh"
 
 namespace celeritas
 {
 //---------------------------------------------------------------------------//
 /*!
  * Axis-aligned bounding box.
  *
- * The bounding box can be constructed in an unassigned state, in which lower
- * and upper cannot be called.
+ * The default bounding box is "degenerate", which encloses no space and
```

Comment:
I don't think "degenerate" is the correct term here. Planes, lines, and points are degenerate cuboids. I would stick to using "null" for describing ((inf, inf, inf), (-inf, -inf, -inf)) as you do in the text of this PR.

Reply:
Good point, I agree completely.